lsp-ui-sideline.el

```diff
@@ -337,13 +344,9 @@ is set to t."
        0)))
 
 (defun lsp-ui-sideline--window-width ()
+  "Calculate the window width only inside the text area."
   (- (min (window-text-width) (window-body-width))
-     (lsp-ui-sideline--margin-width)
-     (or (and (>= emacs-major-version 27)
-              ;; We still need this number when calculating available space
-              ;; even with emacs >= 27
-              (lsp-ui-util-line-number-display-width))
-         0)))
+     (lsp-ui-sideline--margin-width)))
```

Comment:
Move the entire line number calculation to function `lsp-ui-sideline--margin-width`.